### PR TITLE
static loop in: More fee versions for sweepless sweep

### PR DESCRIPTION
**Pull request by sputn1ck**
This PR changes the loopin sweepless sweep
handling, by changing the hardcoded fee messages to
a map of fee versions.

### CODE REVIEW

**Review by hieblmi — APPROVED** [review]
Thanks for the support @sputn1ck 🎖️
Code lgtm, I've done some reordering an naming changes.

File: swapserverrpc/staticaddr.proto [hunk 1]
```diff
@@ -195,44 +195,23 @@ message FetchSweeplessSweepTxResponse {
...
-
-    // The info the server used to generate the high fee partial sweepless tx
-    // sigs.
-    ServerSweeplessSigningInfo high_fee_info = 3;
-
-    // The info the server used to generate the extreme fee partial sweepless tx
-    // sigs.
-    ServerSweeplessSigningInfo extreme_fee_info = 4;
+    // A map of fee rate to the nonces.
+    map<uint64, ServerSweeplessSigningInfo> fee_rate_to_nonces = 2;
```
Comment (bhandras) [inline swapserverrpc/staticaddr.proto:199]: nit: i'd call this `server_signing_info` (and the one in the `PushStaticAddressSweeplessSigsRequest` `client_signing_info`.

File: swapserverrpc/staticaddr.proto [hunk 2]
```diff
@@ -195,44 +195,23 @@ message FetchSweeplessSweepTxResponse {
...
-    ClientSweeplessSigningInfo standard_signing_info = 2;
-
-    // The info the client used to generate the high fee partial sweepless tx
-    // sigs.
-    ClientSweeplessSigningInfo high_fee_signing_info = 3;
-
-    // The info the client used to generate the extreme fee partial sweepless
-    // tx sigs.
-    ClientSweeplessSigningInfo extreme_fee_signing_info = 4;
+    // A map of fee rate to the nonces.
```
Comment (bhandras) [inline swapserverrpc/staticaddr.proto:213]: nit: comment could be updated as it's not just the nonces but also the sigs.

**Review by bhandras — APPROVED** [review]
LGTM, just two nits 🎉

File: swapserverrpc/staticaddr.proto [hunk 3]
```diff
@@ -195,44 +195,23 @@ message FetchSweeplessSweepTxResponse {
...
-    ServerSweeplessSigningInfo standard_fee_info = 2;
-
-    // The info the server used to generate the high fee partial sweepless tx
-    // sigs.
-    ServerSweeplessSigningInfo high_fee_info = 3;
-
-    // The info the server used to generate the extreme fee partial sweepless tx
-    // sigs.
-    ServerSweeplessSigningInfo extreme_fee_info = 4;
+    // A map of fee rate to the nonces.
```
Comment (starius) [inline swapserverrpc/staticaddr.proto:198]: I propose to add units of fee rate to godoc and/or to the name of the field. Also in PushStaticAddressSweeplessSigsRequest `client_signing_info`.

File: staticaddr/loopin/actions.go [hunk 4]
```diff
@@ -804,152 +805,65 @@ func (f *FSM) FetchSignPushSweeplessSweepTxAction(ctx context.Context,
...
 
-	// Sign standard.
-	sweeplessClientSigs, err := f.loopIn.signMusig2Tx(
-		ctx, sweepTx, f.cfg.Signer, sessions, serverNonces,
-	)
-	if err != nil {
-		err = fmt.Errorf("unable to sign sweepless sweep tx: %w", err)
-		return f.HandleError(err)
-	}
+		sessions, nonces, err = f.loopIn.createMusig2Sessions(
```
Comment (starius) [inline staticaddr/loopin/actions.go:827]: I think, resources represented by `sessions` leak in the signer. We should close them. It is not related to this PR, therefore let's add TODO here to clean sessions in the future.

File: staticaddr/loopin/actions.go [hunk 5]
```diff
@@ -804,152 +805,65 @@ func (f *FSM) FetchSignPushSweeplessSweepTxAction(ctx context.Context,
...
-		f.loopIn.createMusig2Sessions(ctx, f.cfg.Signer)
-
-	if err != nil {
-		return f.HandleError(err)
-	}
-	extremelyHighClientNonces, err := toNonces(extremeClientNonces)
-	if err != nil {
-		return f.HandleError(err)
-	}
+	clientResponse := make(map[uint64]*looprpc.ClientSweeplessSigningInfo)
```
Comment (starius) [inline staticaddr/loopin/actions.go:808]: I propose to preallocate the space in the map:
```go
clientResponse := make(
  map[uint64]*looprpc.ClientSweeplessSigningInfo,
  len(fetchResp.FeeRateToNonces),
)
```

**Review by starius — APPROVED** [review]
LGTM! Great improvement!
I added a couple of comments.

**Pull request closed without merging**